wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -586,12 +589,13 @@ instance Arbitrary SignTransactionError where
 
 mkSigner :: NetworkMagic
          -> PassPhrase
-         -> Maybe EncryptedSecretKey
+         -> Maybe Keystore.WalletUserKey
```

Comment:
That's not the right place to make this separation. `mkSigner` should accept an `EncryptedSecretKey`. The caller has to deal with whether it has a key or not.